SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -784,6 +784,504 @@ void CFlowOutput::Set_CpInverseDesign(CSolver *solver, const CGeometry *geometry
 
 }
 
+void CFlowOutput::Add_NearfieldInverseDesignOutput(){
+
+  AddHistoryOutput("EQUIVALENT_AREA",   "CEquiv_Area",  ScreenOutputFormat::SCIENTIFIC, "EQUIVALENT_AREA", "Equivalent area", HistoryFieldType::COEFFICIENT);
+  
+}
+
+void CFlowOutput::Set_NearfieldInverseDesign(CSolver *solver, const CGeometry *geometry, const CConfig *config){
+
+  bool output = true;
+  ofstream EquivArea_file, FuncGrad_file;
+  unsigned short iMarker = 0, iDim;
+  su2double Gamma, auxXCoord, auxYCoord, auxZCoord, InverseDesign = 0.0, DeltaX, Coord_i, Coord_j, jp1Coord, *Coord = nullptr, MeanFuntion,
+  *Face_Normal = nullptr, auxArea, auxPress, Mach, Beta, R_Plane, Pressure_Inf,
+  ModVelocity_Inf, Velocity_Inf[3], factor,
+  jFunction, jp1Function;
+  unsigned long jVertex, iVertex, iPoint, nVertex_NearField = 0, auxPoint,
+  auxDomain;
+  unsigned short iPhiAngle;
+  ofstream NearFieldEA_file; ifstream TargetEA_file;
+
+  su2double XCoordBegin_OF = config->GetEA_IntLimit(0);
+  su2double XCoordEnd_OF = config->GetEA_IntLimit(1);
+
+  unsigned short nDim = geometry->GetnDim();
+  su2double AoA = -(config->GetAoA()*PI_NUMBER/180.0);
+  su2double EAScaleFactor = config->GetEA_ScaleFactor(); // The EA Obj. Func. should be ~ force based Obj. Func.
+
+  Mach  = config->GetMach();
+  Gamma = config->GetGamma();
+  Beta = sqrt(Mach*Mach-1.0);
+  R_Plane = fabs(config->GetEA_IntLimit(2));
+  Pressure_Inf = config->GetPressure_FreeStreamND();
+  Velocity_Inf[0] = config->GetVelocity_FreeStreamND()[0];
+  Velocity_Inf[1] = config->GetVelocity_FreeStreamND()[1];
+  Velocity_Inf[2] = config->GetVelocity_FreeStreamND()[2];
+  ModVelocity_Inf = 0;
+  for (iDim = 0; iDim < 3; iDim++)
+    ModVelocity_Inf += Velocity_Inf[iDim] * Velocity_Inf[iDim];
```

Comment:
```suggestion
  const su2double ModVelocity_Inf = GeometryToolbox::SquaredNorm(3, config->GetVelocity_FreeStreamND());
```
You may need an #include geometry_toolbox.hpp

Reply:
geometry_toolbox.hpp is already included.